vuejs/src/org/jetbrains/vuejs/intentions/expandVModel/VueExpandVModelIntention.kt

```diff
@@ -0,0 +1,70 @@
+package org.jetbrains.vuejs.intentions.expandVModel
+
+import com.intellij.lang.javascript.intentions.JavaScriptIntention
+import com.intellij.openapi.application.WriteAction
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.xml.XmlAttribute
+import com.intellij.psi.xml.XmlElementType
+import org.jetbrains.vuejs.codeInsight.tags.VueElementDescriptor
+
+class VueExpandVModelIntention : JavaScriptIntention() {
+    override fun getFamilyName(): String = "Expand v-model into :value and @input/@change"
+    override fun getText(): String = this.familyName
+    private val validModifiers = setOf("lazy", "number", "trim")
+
+    override fun isAvailable(project: Project, editor: Editor?, element: PsiElement): Boolean {
+        return element.node.elementType == XmlElementType.XML_NAME &&
+                element.parent?.node?.elementType == XmlElementType.XML_ATTRIBUTE &&
+                isValid(element)
+    }
+
+    private fun isValid(element: PsiElement): Boolean {
+        val text = element.text
+        if (!text.startsWith("v-model")) {
+            return false
+        }
+        val modifiers = getModifiers(text)
+        return validModifiers.containsAll(modifiers)
+    }
+
+    private fun getModifiers(text: String): List<String> {
+        val split = text.split('.')
+        return split.subList(1, split.size)
+    }
+
+    override fun invoke(project: Project, editor: Editor?, psiElement: PsiElement) {
+        editor ?: return
+        val parent: PsiElement = psiElement.parent
+        val parentAttr = parent as XmlAttribute
+        val parentTag = parentAttr.parent
+        val directiveText = parent.name
+        val isComponent = parentTag.descriptor is VueElementDescriptor
+        var assignedValue = if (isComponent) "\$event" else "\$event.target.value"
+        var eventName = "@input"
```

Comment:
I think the event name should be taken from the component's `model` property:
```
(parentTag.descriptor as VueElementDescriptor).getModel().event
```
You need to add the `getModel()` function similarly to `getProps()`, but I guess you have to take into account only the first source.

Reply:
` I guess you have to take into account only the first source.`  - what do you mean by that? There can be only one v-model on a component - unless there's a syntax error. In that case, User will still want to transform the model that is under the caret, which is not necessarily the one returned from the descriptor..

Reply:
I meant the implementation for `getModel` function in `VueElementDescriptor`; it has `sources` field, which usually has only one element, but sometimes more. This ambiguity is present, because in same rare cases we are not sure which of the candidates is really used in the template.

Reply:
I misunderstood the meaning of VueElementDescriptor at first. I implemented the method. Used first source but still left defaults (`@input`, `:value`). Those are 90% times the correct ones.